### PR TITLE
Update to Spring Boot 2

### PR DESCRIPTION
**Pull request by galderz**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -52,13 +51,6 @@
                 <type>pom</type>
                 <scope>import</scope>
             </dependency>
-            <dependency>
-                <groupId>org.infinispan</groupId>
```
Comment (slaskawi) [inline pom.xml]: Why did you remove the BOM? It's essential to align this starter to ISPN that you want. Otherwise you're pulling dependencies from Spring (which are not always up to date).

Comment (galderz) [reply pom.xml]: Well, in this case the Infinispan dependency are the ones lagging behind (9.1.0.Final instead of 9.1.7.Final coming from Spring). And the way it's laid out the Spring ones take advantage which is confusing.

Only a set of dependencies should come, which should come from Spring. Adding Infinispan dependencies here it's confusing because which one is used depends on dependency ordering which is very easy to get wrong.

Comment (galderz) [reply pom.xml]: p.s. I'm updating Spring Boot to 9.3.0.Final as we speak...

Comment (galderz) [reply pom.xml]: FYI: https://github.com/spring-projects/spring-boot/pull/13570

Comment (slaskawi) [reply pom.xml]: Yeah, sure - if you prefer to maintain Infinispan dependencies there, I see no problem.

However, maintaining the BOM here has a couple of advantages:
- It's super easy for prod - all you need to do is to replace Spring BOM with RHOAR and Infinispan BOM with JDG and you're done.
- We control the ISPN version directly. This way we can catch newly introduced bugs during the upgrade (since we upgrade ISPN dependency here and can test it out). With BOM provided in Spring, we must update Spring and Infinispan at the same time (since Infinispan dependency is provided by Spring BOM).

Comment (galderz) [reply pom.xml]: Fair point. I'll update to keep Infinispan BOM but MUST come before. I'll document the caveat

**Comment by galderz** [comment]
@slaskawi Updated
